src/semgrep_agent/meta.py

```diff
@@ -35,13 +35,18 @@ def event_name(self) -> str:
 
     @cachedproperty
     def repo(self) -> gitpython.Repo:  # type: ignore
-        repo = gitpython.Repo()
+        repo = gitpython.Repo(".", search_parent_directories=True)
         debug_echo(f"found repo: {repo!r}")
         return repo
 
     @cachedproperty
     def repo_name(self) -> str:
-        return Path.cwd().name
+        repo = gitpython.Repo(".", search_parent_directories=True)
+        return str(os.path.basename(repo.working_tree_dir))
+
+    @cachedproperty
+    def repo_url(self) -> Optional[str]:
+        return os.getenv("REPO_URL")
```

Comment:
For all the baseclass properties that read env variables do we expect notifications backend to still work without these fields? Can we comment somewhere which fields are necessary and which are optional.

Reply:
I just grouped them in line 60 based on required/not. Only "repository" is required, and notifications can work without the rest (although they are nice to have to make the notification more useful). But yes, we now raise a RuntimeError in line 46 if this doesn't get set properly and no commandline arguments are _necessary_ for notifications to work

Reply:
Maybe we should prefix all these env variable names with `SEMGREP_` so it's `SEMGREP_REPO_URL`?

- Someone might have other software (e.g. CI) running that uses the same names
- It's nice when debugging an environment to know what each variable belongs to